server/mediaserver.go

```diff
@@ -308,7 +308,7 @@ func gotRTMPStreamHandler(s *LivepeerServer) func(url *url.URL, rtmpStrm stream.
 				}
 
 				if rpcBcast != nil {
-					SubmitSegment(rpcBcast, seg)
+					go func() { SubmitSegment(rpcBcast, seg) }()
```

Comment:
Make the segment submission asynchronous within a goroutine:
Increases network utilization and decreases the elpased time
between segments, which would otherwise be increasing each time
the upload + transcode took more than 4 seconds. This would
sometimes result in 6 or 7 seconds between segments.

Async submission isn't entirely correct; for example, during
large backlogs, scheduling could cause segments to be delivered
out of order at the transcoder. The most correct thing to do is
to ensure that only one segment is in-flight on the wire at any
given time. However, this quick and easy approach is OK for now.